src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.Value.cs

```diff
@@ -0,0 +1,244 @@
+            public Version Version { get; }
+            public bool Negated { get; }
+
+            public IAbstractAnalysisValue GetNegatedValue()
+                => new RuntimeMethodValue(InvokedMethodName, PlatformName, Version, !Negated);
+
+            public static bool TryDecode(
+                IMethodSymbol invokedPlatformCheckMethod,
+                ImmutableArray<IArgumentOperation> arguments,
+                ValueContentAnalysisResult? valueContentAnalysisResult,
+                INamedTypeSymbol osPlatformType,
+                [NotNullWhen(returnValue: true)] out RuntimeMethodValue? info)
+            {
+                // Accelerators like OperatingSystem.IsPlatformName()
+                if (arguments.IsEmpty)
+                {
+                    var platformName = SwitchPlatformName(invokedPlatformCheckMethod.Name);
+                    if (platformName != null)
+                    {
+                        info = new RuntimeMethodValue(invokedPlatformCheckMethod.Name, platformName, new Version(0, 0), negated: false);
+                        return true;
+                    }
+                }
+                else
+                {
+                    if (TryDecodeRuntimeInformationIsOSPlatform(arguments[0].Value, osPlatformType, out string? osPlatformName))
+                    {
+                        info = new RuntimeMethodValue(invokedPlatformCheckMethod.Name, osPlatformName, new Version(0, 0), negated: false);
+                        return true;
+                    }
+
+                    if (arguments[0].Value is ILiteralOperation literal &&
+                        literal.Type?.SpecialType == SpecialType.System_String &&
+                        literal.ConstantValue.HasValue)
+                    {
+                        // OperatingSystem.IsOSPlatform(string platform)
+                        if (invokedPlatformCheckMethod.Name == IsOSPlatform &&
+                            TryParsePlatformNameAndVersion(literal.ConstantValue.Value.ToString(), out string platformName, out Version? version))
+                        {
+                            info = new RuntimeMethodValue(invokedPlatformCheckMethod.Name, platformName, version, negated: false);
+                            return true;
+                        }
+
+                        // OperatingSystem.IsOSPlatformVersionAtLeast(string platform, int major, int minor = 0, int build = 0, int revision = 0)
+                        if (TryDecodeOSVersion(arguments, valueContentAnalysisResult, out version, 1))
+                        {
+                            info = new RuntimeMethodValue(invokedPlatformCheckMethod.Name, literal.ConstantValue.Value.ToString(), version, negated: false);
+                            return true;
+                        }
+                    }
+                    else if (arguments[0].Value is ILiteralOperation intLiteral && intLiteral.Type?.SpecialType == SpecialType.System_Int32)
```

Comment:
We can pull out “is ILiteralOperation literal” to an outer if to avoid doing the same check twice.